src/CoffeeShop/CoffeeShop.php

```diff
@@ -0,0 +1,30 @@
+<?php
+namespace CoffeeShop;
+use base\Location;
+
+class CoffeeShop extends Location implements IShop{
```

Comment:
When I think about a Coffee Shop I think `it is at a location` that `it is a location` the fact that it implements `ILocation` does not seem to help later on in the implementation.

Reply:
When I wrote this I was thinking that a Coffee Shop is a place/location which is obviously located somewhere which is actually a shop with a name.
Makes sense?

Reply:
As an alternative the composition is used.
Into the CoffeeShop an attribute called location is placed.
One shortcoming is that it tight couples the implementation details between childs and parents.